From 84411871262363200e70fa087d7b21647091a450 Mon Sep 17 00:00:00 2001
From: Ian Campbell <ian.campbell@citrix.com>
Date: Wed, 31 Jul 2013 16:16:16 +0100
Subject: [PATCH] qemu-xen-traditional: allow build without blktap1

I intend this to become optional at the xen build level so it needs to become
optional here. Until the matching Xen patch is applied and exports
CONFIG_BLKTAP1=y|n there should be no change.

Signed-off-by: Ian Campbell <ian.campbell@citrix.com>
Acked-by: Ian Jackson <ian.jackson@eu.citrix.com>
---
 hw/xen_machine_pv.c |    4 ++--
 xen-config-host.h   |    2 +-
 xen-hooks.mak       |    4 +++-
 3 files changed, 6 insertions(+), 4 deletions(-)

diff --git a/hw/xen_machine_pv.c b/hw/xen_machine_pv.c
index c09adbb..e7283b0 100644
--- a/hw/xen_machine_pv.c
+++ b/hw/xen_machine_pv.c
@@ -29,7 +29,7 @@
 #include "xen_backend.h"
 #include "qemu-xen.h"
 
-#ifndef CONFIG_STUBDOM
+#if defined(CONFIG_BLKTAP1) && !defined(CONFIG_STUBDOM)
 #include <hw/xen_blktap.h>
 #endif
 
@@ -47,7 +47,7 @@ static void xen_init_pv(ram_addr_t ram_size, int vga_ram_size,
     CPUState *env;
     uint32_t domid_target;
 
-#if !defined(CONFIG_STUBDOM) && !defined(__NetBSD__)
+#if defined(CONFIG_BLKTAP1) && !defined(CONFIG_STUBDOM) && !defined(__NetBSD__)
     /* Initialize tapdisk client */
     init_blktap();
 #endif
diff --git a/xen-config-host.h b/xen-config-host.h
index 647f6be..3a02021 100644
--- a/xen-config-host.h
+++ b/xen-config-host.h
@@ -18,7 +18,7 @@ extern int domid, domid_backend;
 
 #include "xenctrl.h"
 #include "xenstore.h"
-#if !defined(CONFIG_STUBDOM) && !defined(__NetBSD__)
+#if defined(CONFIG_BLKTAP1) && !defined(CONFIG_STUBDOM) && !defined(__NetBSD__)
 #include "blktaplib.h"
 #endif
 
diff --git a/xen-hooks.mak b/xen-hooks.mak
index 58d61c9..b91400e 100644
--- a/xen-hooks.mak
+++ b/xen-hooks.mak
@@ -47,9 +47,11 @@ CONFIG_AUDIO=
 OBJS += xenfbfront.o
 else
 ifndef CONFIG_NetBSD
-CPPFLAGS+= -I$(XEN_ROOT)/tools/blktap/lib
+ifneq ($(CONFIG_BLKTAP1),n)
+CPPFLAGS+= -DCONFIG_BLKTAP1 -I$(XEN_ROOT)/tools/blktap/lib
 LIBS += -L$(XEN_ROOT)/tools/blktap/lib -lblktap
 OBJS += xen_blktap.o
+endif
 OBJS += tpm_tis.o
 endif
 endif
-- 
1.7.2.5

